components/claim/components/ldap_to_email.tsx

```diff
@@ -0,0 +1,233 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useRef, useState} from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import {t} from 'utils/i18n';
+import {isValidPassword, localizeMessage} from 'utils/utils';
+import {ClaimErrors} from 'utils/constants';
+
+import LoginMfa from 'components/login/login_mfa';
+import LocalizedInput from 'components/localized_input/localized_input';
+
+import {PasswordConfig} from '../claim_controller';
+
+import {AuthChangeResponse} from '@mattermost/types/users';
+
+import {SubmitOptions} from './email_to_ldap';
+
+type Props = {
+    email: string | null;
+    switchLdapToEmail: (ldapPassword: string, email: string, password: string, token: string) => Promise<{data?: AuthChangeResponse; error?: {server_error_id: string; message: string}}>;
+    passwordConfig?: PasswordConfig;
+}
+
+const LDAPToEmail = (props: Props) => {
+    const [passwordError, setPasswordError] = useState<string | JSX.Element>('');
+    const [confirmError, setConfirmError] = useState('');
+    const [ldapPasswordError, setLdapPasswordError] = useState('');
+    const [serverError, setServerError] = useState('');
+    const [showMfa, setShowMfa] = useState(true);
```

Comment:
nit: could you move this to be the last one, so we have every state used for inputs grouped. :bow:

Reply:
Done